### PR TITLE
Consolidate deps

### PR DESCRIPTION
**Pull request by maackle**
### Summary

Here's an ongoing PR to consolidate deps into the workspace. I'm getting the ball rolling, perhaps others want to contribute a few more changes and then we can merge it and replace deps on an ongoing basis.

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain-services/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[package]
```
Comment (neonphog) [inline crates/holochain-services/Cargo.toml:1]: @maackle - I think you based this off the wrong branch

**Comment by jost-s** [comment]
Great, let's do it! I did `rmp-serde` and `lair_keystore` as well as `lair_keystore_api` on a different PR.

You changed the target branch to `conductor services crate`. Shouldn't that remain `develop` and you rebase your local branch on `develop` too?

**Comment by maackle** [comment]
Without going into the reasons for the alternate base, in any case, this PR is now based on develop

**Review by steveej — CHANGES_REQUESTED** [review]
please don't merge this before someone (including me but not necessarily so) is confident that the release-automation can actually bump versions of in-repo crates in the top-level Cargo.toml. IIRC it can't do that today.

**Comment by maackle** [comment]
@steveej that ship has already sailed, via https://github.com/holochain/holochain/pull/3098 which you also commented on. I did a dry release on that PR and it worked, and assumed that was all we needed in order to be confident. If there's another check we should do, we can do it on develop (or find out if today's release succeeds)

**Comment by steveej** [comment]
@maackle 

> that ship has already sailed, via #3098 which you also commented on.

my concern is about "in-repo crates in the top-level Cargo.toml" to quote my previous message. the PR you reference only introduced a crates.io dependency.

**Comment by maackle** [comment]
Ah, I missed that distinction. OK, I'll dry-run this one too.

File: Cargo.toml [hunk 2]
```diff
@@ -1,4 +1,3 @@
-resolver = "2"
```
Comment (ThetaSinner) [inline Cargo.toml:1]: I'd definitely prefer we keep this, is there a reason it's being removed?

Comment (jost-s) [reply Cargo.toml:1]: So do I. It's placed outside of `[workspace]` here and amounts to nothing like this, so it should go back under `[workspace]`.

Comment (maackle) [reply Cargo.toml:1]: Oops, mistake, yes let's leave it.

**Pull request closed without merging**